### ISSUE TITLE
C++: Customization mechanism for the standard library

### ISSUE THREAD

**Issue by artem-smotrakov** [github/codeql#14722]
Hey team,

QL libraries for Java, Golang, and other languages have `Customizations.qll` files that allow applying customizations to the standard libs. For example, `Customizations.qll` is imported by `java.qll`, so any customizations defined there automatically apply to all queries. This is a nice feature if one wants to extend the standard queries, for example, by adding taint sources, sinks, etc that are specific to the codebase being scanned.

I don't see this customization mechanism for the C++ though. Is there a reason why it is not supported for C++? Would it be possible to implement it? It should be as simple as adding a `Customizations.qll` file and importing it in `cpp.qll` unless I am missing something. I am happy to open a pull request for it.

**Comment by aeisenberg** [comment]
Hi @artem-smotrakov, thanks for your question. In the C++ libraries, the file is called `Options.qll`. The behaviour is the same as `Customizations.qll` in other languages.

**Comment by artem-smotrakov** [comment]
Hi @aeisenberg ! Yeah, I saw `Options.qll` but I was not sure the behaviour is the same.

In other languages, for example in Java, I see `Customizations.qll` is imported on the top of `java.qll`, but `cpp.qll` doesn't import `Options.qll` (only `DefaultOptions.qll`). Not sure if that matters though. If I want to add a new taint source for all queries, would it work if I added it to `Options.qll`? 

Also, I see `Customizations.qll` files are empty, but `Options.qll` is not. In practice, it may be convenient to have a custom `Customizations.qll` somewhere with all customizations and just copy it to the `language/ql/lib` wherever necessary (in CI/CD for example). Since `Options.qll` is not empty, looks like the customizations needs to be added to the bottom of this file. It is not a big deal, but having the same customization mechanism for all languages would be a bit more convenient :)
